### PR TITLE
Port to ROS Noetic

### PR DESCRIPTION
**Pull request by mjeronimo**
# PR Details
## Description

Changes required for port to ROS Noetic (Python3, Ubuntu 20.04, etc.)

## Motivation and Context

Updates for Python scripts, dependencies, etc. 

## How Has This Been Tested?

Run unit tests. 

## Types of changes

- [X] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -52,6 +52,10 @@ FROM base-image AS install
 RUN mkdir ~/carma_ws
 COPY --from=source-code --chown=carma /home/carma/src /home/carma/carma_ws/src
 
+# add astuff messages
+WORKDIR /home/carma/carma_ws/src
+RUN git clone https://github.com/astuff/astuff_sensor_msgs -b melodic
```
Comment (msmcconnell) [inline Dockerfile:57]: I think this clone would be better placed in the docker/checkout.sh script to match the pattern our team has been using.

File: carma-platform.repos [hunk 2]
```diff
@@ -1,48 +1,60 @@
 repositories:
+  carma/src/astuff_sensor_msgs:
+    type: git
+    url: https://github.com/astuff/astuff_sensor_msgs.git
+    version: melodic
+  autoware.ai:
+    type: git
+    url: https://github.com/mjeronimo/autoware.ai.git
```
Comment (msmcconnell) [inline carma-platform.repos:8]: Well want to point this at our autoware.ai fork once that is merged. Same with the other links in this file

File: docker/checkout.bash [hunk 3]
```diff
@@ -40,11 +40,11 @@ cd ${dir}/src
...
       git clone --depth=1 https://github.com/usdot-fhwa-stol/carma-msgs.git --branch develop
       git clone --depth=1 https://github.com/usdot-fhwa-stol/novatel_gps_driver.git --branch develop
-      git clone --depth=1 https://github.com/usdot-fhwa-stol/carma-utils.git --branch develop
+      git clone --depth=1 https://github.com/mjeronimo/carma-utils.git --branch feature/ros-noetic-port
       git clone --depth=1 https://github.com/usdot-fhwa-stol/carma-messenger.git --branch develop
 else
       git clone --depth=1 https://github.com/usdot-fhwa-stol/carma-msgs.git --branch develop
       git clone --depth=1 https://github.com/usdot-fhwa-stol/novatel_gps_driver.git --branch develop
-      git clone --depth=1 https://github.com/usdot-fhwa-stol/carma-utils.git --branch develop
+      git clone --depth=1 https://github.com/mjeronimo/carma-utils.git --branch feature/ros-noetic-port
```
Comment (msmcconnell) [inline docker/checkout.bash:48]: Need to remember to update these links once the respective repos are merged

File: docker/install.sh [hunk 4]
```diff
@@ -19,11 +19,10 @@
 
 set -ex
 
-source /opt/ros/kinetic/setup.bash
 source /opt/autoware.ai/ros/install/setup.bash --extend
 
 cd ~/carma_ws
-./src/carma-platform/carma_build.bash -c ~/carma_ws -a /opt/autoware.ai/ -x -m "-DCMAKE_BUILD_TYPE=Release"
+./src/carma-platform/carma_build.bash -c ~/carma_ws -a /opt/autoware.ai/ -x -m "-DCMAKE_BUILD_TYPE=Debug"
```
Comment (msmcconnell) [inline docker/install.sh:25]: This should be kept as Release

File: engineering_tools/RouteCreation_CSV2Yaml.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
Comment (msmcconnell) [inline engineering_tools/RouteCreation_CSV2Yaml.py:1]: If this is running in ubuntu 20.04 do we need to specify python3 ?

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: carma-platform.repos (hunk 2 above)
Comment (mjeronimo) [reply carma-platform.repos:8]: Yes, I've added a tasks to make sure this doesn't get missed. We can integrate carma-base and autoware.ai first and then do this PR last, updating the URLs before merging.

File: docker/checkout.bash (hunk 3 above)
Comment (mjeronimo) [reply docker/checkout.bash:48]: @msmcconnell I'll update. I suppose these should all be "noetic/develop", correct? Same goes for all other checkout.bash files in other repos.

File: docker/install.sh (hunk 4 above)
Comment (mjeronimo) [reply docker/install.sh:25]: Yes. We currently have a crash that we need to debug w/ the Release version. Will debug that crash and fix this before merging.

File: engineering_tools/RouteCreation_CSV2Yaml.py (hunk 5 above)
Comment (mjeronimo) [reply engineering_tools/RouteCreation_CSV2Yaml.py:1]: Yes, that's the common practice. Also, 'python' is not available, by default:

```
$ docker run -it ros:noetic
root@ef226c5d7aa2:/# python
bash: python: command not found
```

File: docker/checkout.bash (hunk 3 above)
Comment (msmcconnell) [reply docker/checkout.bash:48]: correct

File: Dockerfile (hunk 1 above)
Comment (mjeronimo) [reply Dockerfile]: Fixed.

File: docker/install.sh (hunk 4 above)
Comment (mjeronimo) [reply docker/install.sh]: Crash fixed and script changed back to Release.

**Review by msmcconnell — APPROVED** [review]

**Pull request merged**